doc/reference/reference_rock/vshard/vshard_ref.rst

```diff
@@ -641,6 +655,44 @@ Router public API
 
     Force wakeup of the bucket discovery fiber.
 
+.. _router_api-discovery_set:
+
+.. function:: vshard.router.discovery_set(mode)
+
+    Turn on/off the background discovery fiber used by the router to
+    find buckets.
+
+    :param mode: working mode of a discovery fiber. There are three modes: ``on``,
+                 ``off`` and ``once``
+
+    When the mode is ``on`` (default), the discovery fiber works during all the lifetime
+    of the router. Even after all buckets are discovered, it will
+    still come to storages and download their buckets with some big
+    period (`DISCOVERY_IDLE_INTERVAL <https://github.com/tarantool/vshard/blob/master/vshard/consts.lua>`_).
+    This is useful if the bucket topology changes often and the number of
+    buckets is not big. The router will keep its route table up to
+    date even when no requests are processed.
+
+    When the mode is ``off``, discovery is disabled completely.
+
+    When the mode is ``once``, discovery will start, find the locations of
+    all buckets, and then the discovery fiber is terminated. This
+    is good for a large bucket count and for clusters, where rebalancing is rare.
+
+    The method is good to enable/disable discovery after the router is
+    already started, but discovery is enabled by default. You may want
+    to never enable it even for a short time -- then specify the
+    ``discovery_mode`` option in the :ref:`configuration <cfg_basic-discovery_set>`.
```

Comment:
let's add `discovery-mode` to the list of options here, like we do for `discovery_set`. otherwise the reference feels "patchy".

Reply:
my mistake. There is the only cfg option -- discovery_mode. Fixed in 3d2f7ac